api4/team_test.go

```diff
@@ -1449,6 +1449,20 @@ func TestInviteUsersToTeam(t *testing.T) {
 			}
 		}
 	}
+
+	utils.Cfg.TeamSettings.RestrictCreationToDomains = "@example.com"
```

Comment:
Just a style note (and an intro to one of Go's features since you said you were new to it), you can actually reset the settings value in a way that's a bit easier and more error-proof if something happened to return early. If you look at other tests, you'll see a block like:
```go
restrictCreationToDomains := utils.Cfg.TeamSettings.RestrictCreationToDomains
defer func() {
	utils.Cfg.TeamSettings.RestrictCreationToDomains = restrictCreationToDomains
}()
utils.Cfg.TeamSettings.RestrictCreationToDomains = "@example.com"
```

Since that's using a deferred function, it'll get the old value of the setting and then reset it right before the function returns or crashes for any reason. It'll also make sure to reset it to the original value, even if that value was changed.

Could you change this part here to match that?